Library/Homebrew/test/tap_auditor_spec.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+require "tap_auditor"
+
+RSpec.describe Homebrew::TapAuditor do
+  subject(:auditor) { described_class.new(tap, strict: false) }
+
+  let(:tap) { Tap.fetch("homebrew", "foo") }
+  let(:tap_path) { tap.path }
+
+  before do
+    tap_path.mkpath
+    tap.clear_cache
+  end
+
+  describe "#audit" do
+    context "with cask_renames.json" do
+      let(:cask_renames_path) { tap_path/"cask_renames.json" }
+      let(:cask_path) { tap_path/"Casks"/"newcask.rb" }
+
+      before do
+        cask_path.dirname.mkpath
+        cask_path.write <<~RUBY
+          cask "newcask" do
+            version "1.0"
+            url "https://brew.sh/newcask-1.0.dmg"
+            name "New Cask"
+            homepage "https://brew.sh"
+          end
+        RUBY
+      end
+
+      it "detects .rb extension in old cask name (key)" do
+        cask_renames_path.write JSON.pretty_generate({
+          "oldcask.rb" => "newcask",
+        })
+
+        auditor.audit
```

Comment:
Use `before` and `subject` to DRY this up

Reply:
@loganrosen Still waiting for this comment to be addressed. Shout when done and I'll review.

Reply:
Apologies – I've DRYed up the tests further.